yaml.go

```diff
@@ -545,7 +545,7 @@ func getStructInfo(st reflect.Type) (*structInfo, error) {
 
 		info := fieldInfo{Num: i}
 
-		tag := field.Tag.Get("yaml")
+		tag := field.Tag.Get(YAML_STRUCT_TAG)
```

Comment:
I encounter the same problem when parsing k8s resources.
Does it make sense to make it fall back on json tags if yaml tags are not available?